app-ui/src/Dataset.tsx

```diff
@@ -139,11 +140,22 @@ function DatasetView() {
     event.preventDefault()
   }
 
-  // if the dataset is not loaded yet, display a loading message
-  if (!dataset) {
-    return <div className='empty'>
-      <h3>Loading...</h3>
-    </div>
+  // if the dataset is not found, display a message
+  if (datasetError) {
+    if ([401, 404].includes(datasetError.status)) {
```

Comment:
Will it be better if we use this: 

```
function isClientError(statusCode: number): boolean {
  return statusCode >= 400 && statusCode < 500;
}
```

We only explicitly throw 401, 403 and 404 iirc. This catches all client side errors in the same bucket.

Reply:
Or we can let it be and see if we need this.